src/test/java/com/google/sps/api/GmailConfigurationTest.java

```diff
@@ -0,0 +1,163 @@
+
+/** Tests querying users and sending emails from an authorized account.*/
+@RunWith(JUnit4.class)
+public final class GmailConfigurationTest {
+
+  private static final String TO = "test@google.com";
+  private static final String SUBJECT = "test";
+  private static final String CONTENT = "<h1>test</h1>";
+
+  private static final String NAME = "test";
+  private static final String EMAIL = "test@google.com";
+  private static final String COLLEGE_A = "000000";
+  private static final String COLLEGE_B = "000001";
+
+  private static final String SUCCESS_MSG = "Successfully sent a new post email to: ";
+  private static final String LOGGER_NAME = "com.google.sps.api";
+
+  @Mock private static Gmail mGmail;
+  @Mock private static MimeMessage mMimeMessage;
+  @Mock private static GmailAPI mGmailAPI;
+  @Mock private static PreparedQuery mpq;
+
+  private static LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private static DatastoreService datastore;
+  private static GmailConfiguration gmailConfiguration;
+  private static MemoryAppender memoryAppender;
+  private static Post post;
+
+  @Before
+  public void setUp() throws Exception {
+
+    helper.setUp();
+
+    Logger logger = (Logger) LoggerFactory.getLogger(LOGGER_NAME);
+    memoryAppender = new MemoryAppender();
+    memoryAppender.setContext((LoggerContext) LoggerFactory.getILoggerFactory());
+    logger.setLevel(Level.DEBUG);
+    logger.addAppender(memoryAppender);
+    memoryAppender.start();
+
+    post = new Post();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    gmailConfiguration = new GmailConfiguration();
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @After
+  public void tearDown() {
+
+    memoryAppender.reset();
+    memoryAppender.stop();
+    helper.tearDown();
+  }
+
+  @Test
+  public void userAttendingCollege() throws IOException {
+    // Tests notifying users when the query for a college has users.
+    
+    Entity userEntity = new Entity("User", EMAIL);
+    userEntity.setProperty("name", NAME);
+    userEntity.setProperty("college", COLLEGE_A);
+    datastore.put(userEntity);
+
+    Filter universityFilter = new FilterPredicate("college", FilterOperator.EQUAL, COLLEGE_A);
+    Query q = new Query("User").setFilter(universityFilter);
+    PreparedQuery pq = datastore.prepare(q);
+
+    when(mpq.asIterable()).thenReturn(pq.asIterable());
+    GmailConfiguration.notifyUsers(COLLEGE_A, post);
+
+    Assert.assertEquals(1, memoryAppender.countEventsForLogger(LOGGER_NAME));
+    Assert.assertTrue(memoryAppender.contains(SUCCESS_MSG+EMAIL, Level.INFO));    
+    memoryAppender.reset();
+  }
+
+  @Test
+  public void noUserAttendingCollege() throws IOException {
+    // Tests notifying users when the query for a college has no users.
+
+    GmailConfiguration.notifyUsers(COLLEGE_B, post);
+    Assert.assertEquals(0, memoryAppender.countEventsForLogger(LOGGER_NAME));
```

Comment:
In addition to this, there should be a way to verify that certain calls are made e.g. https://stackoverflow.com/questions/14889951/how-to-verify-a-method-is-called-two-times-with-mockito-verify

Reply:
Same as below as well... We can verify that some calls were called -- it's not ideal since in tests we don't want to test the internals of a method otherwise, it would be brittle whenever the internal details are changed. But this is the best we could do.

Reply:
Including!